source/MaterialXCore/Interface.cpp

```diff
@@ -65,6 +65,95 @@ NodePtr PortElement::getConnectedNode() const
     return graph ? graph->getNode(getNodeName()) : nullptr;
 }
 
+void PortElement::setConnectedOutput(ConstOutputPtr output)
```

Comment:
I forgot to leave a comment. This isn't exactly "safe" as if there is a connection to a node only `nodename` is note cleared leave a possible invalid input->node or output->node connection to the first output on the upstream node.

Reply:
Also moving this to PortElement, I'm not sure if there is specific configurations of output->output connections which should not be allowed (type is not checked here).

Reply:
@kwokcb Feel free to propose improvements to this, and my own recommendation would be to check for invalid configurations at document validation time (e.g. in Element::validate and its specializations), rather than aiming to enforce correctness at document construction time.